docs/bos/queryapi/how-works.md

```diff
@@ -0,0 +1,45 @@
+---
+id: how-it-works
+title: How QueryAPI works
+sidebar_label: How it works
+---
+
+QueryApi is a streaming indexer implementation that executes queries written by developers on the NEAR blockchain.
+QueryApi allows hosted execution of complex queries (ones that can’t be answered by a [simple RPC](../../5.api/rpc/introduction.md) or [Enhanced API](https://docs.pagoda.co/api) call) and hosting of the data output of those queries using GraphQL endpoints.
+
+
+## Components involved 
+
+The QueryApi implementation integrates many different components in a single and streamlined solution.
+In a high-level overview, the main components are:
+
+:::info Components
+`NEAR Protocol` -> `NEAR Lake` -> `Coordinator` -> `Database` -> `API`
+:::
+
+
+### Detailed overview
+
+An in-depth, detailed overview of the QueryApi components:
+
+[![QueryAPI](/docs/qapi-components.png)](/docs/qapi-components.png)
+
+
+### Description
+
+- **Protocol:** the underlying NEAR Layer-1 Blockchain, where data `Blocks` and `Chunks` are produced.
+- **NEAR Lake:** an indexer built on top of the [NEAR Lake Framework](../../1.concepts/3.advanced/near-lake-framework.md) to watch the Layer-1 network and store all the events as JSON files on AWS S3. Changes are indexed as new `Blocks` arrive.
```

Comment:
```suggestion
- **NEAR Lake:** an indexer which watches the Layer-1 network and stores all the events as JSON files on AWS S3. Changes are indexed as new `Blocks` arrive.
```
`Near Lake Framework` is actually the libraries which consumer from Near Lake/S3